### PR TITLE
[NO SQUASH] Move to C++17 and remove Boost dependency

### PR DESCRIPTION
**Pull request by mrkubax10**
This PR includes following changes:

- [x] Use next C++ standard (C++17)
- [x] Remove Boost dependency because most of required functionality is part of STL since C++17

### CODE REVIEW

File: src/supertux/gameconfig.cpp [hunk 1]
```diff
@@ -509,4 +509,15 @@ Config::save()
...
 
+
+bool
+Config::is_christmas() const
+{
+  std::time_t time = std::time(0);
+  std::tm* now = std::localtime(&time);
+
+  /* Activate Christmas mode from Dec 6th until Dec 31st. */
+  return now->tm_mday >= 6 && now->tm_mon == 11;
```
Comment (tobbi) [inline src/supertux/gameconfig.cpp:520]: Is `tm_mon` 0-based?

Comment (Vankata453) [reply src/supertux/gameconfig.cpp:520]: Yes.

File: src/supertux/main.cpp [hunk 2]
```diff
@@ -641,14 +640,9 @@ Main::run(int argc, char** argv)
...
   // - targets where libstdc++ uses its generic locales code (https://gcc.gnu.org/legacy-ml/libstdc++/2003-02/msg00345.html)
-  // NOTE: when moving to C++ >= 17, keep the try-catch block, but use std::locale:global(std::locale(""));
-  //
-  // This should not be necessary on *nix, so only try it on Windows.
   try
   {
-    std::locale::global(boost::locale::generator().generate(""));
-    // Make boost.filesystem use it
-    boost::filesystem::path::imbue(std::locale());
+    std::locale::global(std::locale::global(std::locale("")));
```
Comment (tobbi) [inline src/supertux/main.cpp:645]: Is there a duplicate `std::locale::global` in there?

**Comment by MatusGuy** [comment]
W pull request i need this for android boost is a pain

**Comment by MatusGuy** [comment]
> Can I merge this?

Honestly idk what u r waiting for. Vankata suspected it had to do with other pull requests? Which ones? Idk

**Pull request merged**